build/opt.go

```diff
@@ -32,9 +32,49 @@ import (
 	"github.com/moby/buildkit/util/entitlements"
 	"github.com/opencontainers/go-digest"
 	"github.com/pkg/errors"
+	"github.com/tonistiigi/fsutil"
 )
 
-func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Options, bopts gateway.BuildOpts, configDir string, pw progress.Writer, docker *dockerutil.Client) (solveOpt *client.SolveOpt, release func(), err error) {
+type solveOpt struct {
+	client.SolveOpt
+}
+
+func (so *solveOpt) setLocalMount(name, root string, addVCSLocalDir func(key, dir string, so *solveOpt)) error {
+	lm, err := newFS(root)
+	if err != nil {
+		return err
+	}
+	if so.LocalMounts == nil {
+		so.LocalMounts = map[string]fsutil.FS{}
+	}
+	so.LocalMounts[name] = lm
+	if addVCSLocalDir != nil {
+		addVCSLocalDir(name, lm.root, so)
+	}
+	return nil
+}
+
+type fs struct {
```

Comment:
this wrapper doesn't seem to do much. It is only called once, stores a property and then same caller reads out the property. `setLocalMount` could just set the path directly.